sources/api/certdog/src/main.rs

```diff
@@ -0,0 +1,401 @@
+    trusted_bundle.retain(|pem| !distrusted.contains(pem));
+
+    // Write a PEM formatted bundle from trusted certificates
+    fs::write(
+        &trusted_store,
+        pems_to_string(&trusted_bundle, add_comment)?,
+    )
+    .context(error::UpdateTrustStore)?;
+
+    Ok(())
+}
+
+/// Returns a list with Pem objects from a PemIterator
+fn pems_from_iter<R>(iter: x509_parser::pem::PemIterator<R>) -> Result<Vec<x509_parser::pem::Pem>>
+where
+    R: BufRead + Seek,
+{
+    let mut vec: Vec<x509_parser::pem::Pem> = Vec::new();
+    for pem in iter {
+        let pem = pem.context(error::ParsePEM)?;
+        vec.push(pem);
+    }
+    Ok(vec)
+}
+
+/// Concatenates all the PEM objects as a single PEM bundle
+fn pems_to_string(pems: &Vec<x509_parser::pem::Pem>, add_comment: bool) -> Result<String> {
+    let mut out = String::new();
+
+    for pem in pems {
+        writeln!(out, "{}", pem_to_string(pem, add_comment)?).context(error::WritePEMString)?;
+    }
+
+    Ok(out)
+}
+
+/// Transforms a PEM object into a PEM formatted string
+fn pem_to_string(pem: &x509_parser::pem::Pem, add_comment: bool) -> Result<String> {
+    let mut out = String::new();
+
+    // If provided, a comment will be added before the PEM formatted string to
+    // identify the certificate. This is useful for debugging purposes
+    if add_comment {
+        if let Some(comment) = comment_for_pem(&pem)? {
+            writeln!(out, "# {}", comment).context(error::WritePEMString)?;
+        }
+    }
+
+    writeln!(out, "{} {}{}", PEM_HEADER, pem.label, PEM_SUFFIX).context(error::WritePEMString)?;
+    let encoded = base64::encode(&pem.contents);
+    let bytes = encoded.as_bytes();
+    for chunk in bytes.chunks(64) {
+        let chunk = String::from_utf8_lossy(chunk);
+        writeln!(out, "{}", chunk).context(error::WritePEMString)?;
+    }
+    writeln!(out, "{} {}{}", PEM_FOOTER, pem.label, PEM_SUFFIX).context(error::WritePEMString)?;
+
+    Ok(out)
+}
+
+/// Returns a string from the common name, organizational unit or organization
+/// fields in the certificate
+fn comment_for_pem(pem: &x509_parser::pem::Pem) -> Result<Option<String>> {
+    let cert = pem.parse_x509().context(error::ParseX509Certificate)?;
+    let subject = cert.tbs_certificate.subject;
+
+    if let Some(common_name) = subject.iter_common_name().next() {
+        if let Ok(common_name_str) = common_name.as_str() {
+            return Ok(Some(common_name_str.to_string()));
+        }
+    } else if let Some(organizational_unit) = subject.iter_organizational_unit().next() {
+        if let Ok(organizational_unit_str) = organizational_unit.as_str() {
+            return Ok(Some(organizational_unit_str.to_string()));
+        }
+    } else if let Some(organization) = subject.iter_organization().next() {
+        if let Ok(organization_str) = organization.as_str() {
+            return Ok(Some(organization_str.to_string()));
+        }
+    }
+
+    Ok(None)
+}
```

Comment:
Is this what's required to mimic the default comments in our CA certificates bundle? It seems like a lot of work to produce output that might not be satisfactory in all cases.

```suggestion
fn comment_for_pem(pem: &x509_parser::pem::Pem) -> Result<Option<String>> {
    let cert = pem.parse_x509().context(error::ParseX509Certificate)?;
    let subject = cert.tbs_certificate.subject;
    let comment = subject                                                                                                                              
        .iter_common_name()                                                                                                                            
        .chain(subject.iter_organizational_unit())                                                                                                     
        .chain(subject.iter_organization())                                                                                                            
        .next();                                                                                                                                       
    Ok(comment                                                                                                                                         
        .and_then(|c| c.as_str().ok())                                                                                                                 
        .and_then(|c| Some(c.to_string())))
}
```